fix(android,fileIO) : remove file check that prevent file reading on android

arthur740212 opened the pull request:
`File.Exist` will always return false for android files, thus this is modified.

The UNITY_EDITOR `Write` can be isolated because `Write` is only meant to happen on editor play mode.

The file check for other platforms could be omitted because they are handled in `ReadAllText` function of their respective platform.

arthur740212 committed:
fix(android,fileIO) : remove file check that prevent file reading on android

paulhazen commented on Assets/Plugins/Source/Core/Config.cs:
```diff
@@ -280,22 +280,16 @@ protected virtual async Task ReadAsync()
-
-            if (configFileExists)
+            
+#if UNITY_EDITOR // This should only happen in Editor Runtime (Play Mode)
+            if (!File.Exists(FilePath)) 
```
During runtime when not in the editor - how do Config files get read?

I think most of them utilize `ReadAsync`, but I think that's just a coincidence.

arthur740212 replied:
During runtime on any platform (including editor), it does use this `Read()`
The comment here is talking about the `Write`.
I'll add a couple words to make it clear.

ReadAsync is used in editor when not in runtime (i.e. in EditorsWindows)

arthur740212 committed:
chore(fileIO) : remove redundant code

The Create is Creating a directory instead of creating the file, and without anything else done with configFile, the Config is still getting created, so this part could be removed

paulhazen reviewed (changes requested):
My main concern is that I was fairly certain that the `Read()` function on the `Config` class _is_ utilized within contexts where the editor is not running.

Regardless of that factor - changes made to `Read()` should also be replicated in `ReadAsync()`.

Is it possible that a better solution would be to never use `File.Exists`, but instead implement a platform-aware method within `FileUtility` that can determine if a file exists on any platform (I believe only Android would need to have a unique implementation).

arthur740212 commented:
> My main concern is that I was fairly certain that the `Read()` function on the `Config` class _is_ utilized within contexts where the editor is not running.
> 
To be precise.
`Read()` is for Player mode (including player mode on editor).
`ReadAsync()` is for editor mode (in other words, in editor but not in player mode).

> Regardless of that factor - changes made to `Read()` should also be replicated in `ReadAsync()`.
> 
`ReadAsync()` does not need these changes because it only runs in the editor, which means File.Exist will have expected behavior.

> Is it possible that a better solution would be to never use `File.Exists`, but instead implement a platform-aware method within `FileUtility` that can determine if a file exists on any platform (I believe only Android would need to have a unique implementation).
We can, but in this case wouldn't need to.

arthur740212 commented:
Because `ReadAsync` is only called in Editor Windows (kudos for handling those). 
I am thinking if there is a way to separate the two and prevent cross usage. The easiest way is simply doing an #if UNITY_EDITOR around `ReadAsync`

andrew-hirata-playeveryware commented on Assets/Plugins/Source/Core/Config.cs:
```diff
@@ -280,22 +280,16 @@ protected virtual async Task ReadAsync()
-            }
-            else
+            
+#if UNITY_EDITOR // This should only happen in Editor Runtime (Play Mode)
```
We should leave a comment on _why_ this should only happen in editor mode.

arthur740212 committed:
chore(comment) : update comment to better describe config write use cases
